contracts/core/fee-calculators/PerformanceFeeCalculator.sol

```diff
@@ -0,0 +1,383 @@
+        view
+    {
+        require(
+            parameters.profitFeePercentage <= maximumProfitFeePercentage,
+            "PerformanceFeeCalculator.validateFeeParameters: Profit fee exceeds maximum."
+        );
+
+        require(
+            parameters.streamingFeePercentage <= maximumStreamingFeePercentage,
+            "PerformanceFeeCalculator.validateFeeParameters: Streaming fee exceeds maximum."
+        );
+
+        ScaleValidations.validateMultipleOfBasisPoint(parameters.profitFeePercentage);
+        ScaleValidations.validateMultipleOfBasisPoint(parameters.streamingFeePercentage);
+
+        require(
+            parameters.highWatermarkResetFrequency >= parameters.profitFeeFrequency,
+            "PerformanceFeeCalculator.validateFeeParameters: Fee collection frequency must exceed highWatermark reset."
+        );
+    }
+
+    /**
+     * Calculates streaming fee by multiplying streamingFeePercentage by the elapsed amount of time since the last fee
+     * was collected divided by one year in seconds, since the fee is a yearly fee.
+     *
+     * @return uint256        Streaming fee
+     */
+    function calculateStreamingFee()
+        internal
+        view
+        returns(uint256)
+    {
+        uint256 timeSinceLastFee = block.timestamp.sub(lastStreamingFeeTimestamp(msg.sender));
+
+        // Streaming fee is streaming fee times years since last fee
+        return timeSinceLastFee.mul(streamingFeePercentage(msg.sender)).div(ONE_YEAR_IN_SECONDS);
+    }
+
+    /**
+     * Calculates profit fee net of streaming fee. Value of rebalancing Set is determined then streaming fee subtracted,
+     * to get postStreamingValue. This value is compared to the highWatermark, if greater than highWatermark multiply by
+     * profitFeePercentage and divide by rebalancingSetValue to get inflation from profit fees. If postStreamingValue does
+     * not exceed highWatermark then return 0.
+     *
+     * @return uint256        Streaming fee
+     */
+    function calculateProfitFee(uint256 _streamingFee)
+        internal
+        view
+        returns(uint256)
+    {
+        // Calculate post streaming value and get high watermark
+        uint256 rebalancingSetValue = calculateRebalancingSetValue(msg.sender);
+        uint256 postStreamingValue = rebalancingSetValue.mul(ONE_HUNDRED_PERCENT.sub(_streamingFee)).deScale(); 
+        uint256 highWatermark = highWatermark(msg.sender);
+
+        // Subtract high watermark from post streaming fee value, unless less than 0 set to 0
+        uint256 gainedValue = postStreamingValue > highWatermark ? postStreamingValue.sub(highWatermark) : 0;
+
+        // Determine percent fee in terms of current rebalancing Set value
+        return gainedValue.mul(profitFeePercentage(msg.sender)).div(rebalancingSetValue);
+    }
+
+    /**
+     * Calculates value of RebalancingSetToken.
+     *
+     * @return uint256        Streaming fee
+     */
+    function calculateRebalancingSetValue(
```

Comment:
Since, I feel this would be reused a lot, we should move this into its own library. (Would need to pass in the oracleWhitelist as a parameter if so)